File: 1_network_demo.js
const { getDriver } = require('./driver');
const { Command } = require('selenium-webdriver/lib/command');

(async () => {
    driver = getDriver();
    const cmdName = 'setNetworkConditions';
    const args = {'offline': true, 'latency': 250, 'download_throughput': 1, 'upload_throughput': 1, 'throughput': 1};
    try {
        const cmd = new Command(cmdName);
        cmd.setParameter('network_conditions', args);
        await driver.schedule(cmd, 'test');
        await driver.schedule(cmd);
        await driver.get('https://testingstage.com/');
    } catch (err) {
        // pass
        console.log(err)
    } finally {
        await new Promise(r => setTimeout(r, 7000));
        await driver.close();
    }

})();

File: 1_network_demo.py
import time

from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.common.keys import Keys

from driver import get_driver


driver = get_driver()
cmd = 'Network.emulateNetworkConditions'
args = {'offline': True, 'latency': 250, 'downloadThroughput': 1, 'uploadThroughput': 1, 'connectionType': 'none'}
driver.execute('executeCdpCommand', {'cmd': cmd, 'params': args})
try:
    driver.get('https://testingstage.com/')
except:
    ActionChains(driver).send_keys(Keys.SPACE).perform()
    pass
time.sleep(7)
driver.close()
